target/linux/ath79/dts/qca9558_allnet_all-wap02860ac.dts

```diff
@@ -75,67 +77,21 @@
 	};
 };
 
-&pcie0 {
-	status = "okay";
-
-	wifi@0,0 {
-		compatible = "qcom,ath10k";
-		reg = <0 0 0 0 0>;
+&partitions {
+	art: partition@ff0000 {
+		label = "art";
+		reg = <0xff0000 0x010000>;
+		read-only;
 	};
 };
 
-&pll {
-	clocks = <&extosc>;
-};
-
-&spi {
+&pcie0 {
 	status = "okay";
 
-	flash@0 {
-		compatible = "jedec,spi-nor";
-		reg = <0>;
-		spi-max-frequency = <25000000>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "u-boot";
-				reg = <0x000000 0x040000>;
-				read-only;
-			};
-
-			partition@40000 {
-				label = "u-boot-env";
-				reg = <0x040000 0x010000>;
-			};
-
-			partition@50000 {
-				label = "custom";
-				reg = <0x050000 0x050000>;
-				read-only;
-			};
-
-			partition@a0000 {
-				compatible = "denx,uimage";
-				label = "firmware";
-				reg = <0x0a0000 0xcd0000>;
-			};
-
-			partition@d70000 {
-				label = "failsafe";
-				reg = <0xd70000 0x280000>;
-				read-only;
-			};
-
-			art: partition@ff0000 {
-				label = "art";
-				reg = <0xff0000 0x010000>;
-				read-only;
-			};
-		};
+	wifi@0,0,0 {
+		compatible = "qcom,ath10k";
+		reg = <0x0 0 0 0 0>;
+		qca,no-eeprom;
```

Comment:
Drop this property, ath10k doesn't understand it.

Reply:
are you sure? the reason I put this is because you had this in the caldata script which suggests the caldata is on flash instead of eeprom...

specifically the art partition from
0x5000
to
0x5844

```
	allnet,all-wap02860ac|\
	glinet,gl-x750)
		caldata_extract "art" 0x5000 0x844
		ath10k_patch_mac $(macaddr_add $(mtd_get_mac_binary art 0x0) +2)
		;;
```

what does kernel log say exactly?

also if you are not sure, you can dump the art partition and upload it for us to see
(I think it's mtd7, verify with `cat /proc/mtd`)

`hexdump -C /dev/mtd7`

Reply:
There's no point in what You're trying discuss here. The point is, even if You add this property here, it will be ignored by the driver. Run `grep -R 'no-eeprom' drivers/net/wireless/ath` in root of Linux source tree, You'll see that only driver which understands this is ath9k.

Reply:
I just checked the ath10k docs

Sorry, all this time I did not realize 😆